client/src/rr/dnssec/signer.rs

```diff
@@ -416,18 +416,39 @@ impl Signer {
         return Ok((ac & 0xFFFF) as u16); // this is unnecessary, no?
     }
 
-    fn hash_message(&self, message: &Message) -> DnsSecResult<Vec<u8>> {
+    fn hash_message(&self, message: &Message, pre_sig0: &Record) -> DnsSecResult<Vec<u8>> {
         // TODO: should perform the serialization and sign block by block to reduce the max memory
         //  usage, though at 4k max, this is probably unnecessary... For AXFR and large zones, it's
         //  more important
         let mut buf: Vec<u8> = Vec::with_capacity(512);
+        let mut buf2: Vec<u8> = Vec::with_capacity(512);
 
         {
-            let mut encoder: BinEncoder = BinEncoder::with_mode(&mut buf, EncodeMode::Signing);
-            message.emit(&mut encoder).unwrap(); // coding error if this panics (i think?)
+            let mut encoder: BinEncoder = BinEncoder::with_mode(&mut buf, EncodeMode::Normal);
+            match pre_sig0.rdata() {
+                &RData::SIG(ref pre_sig0) =>
+                    assert!(sig::emit_pre_sig(&mut encoder,
```

Comment:
`hash_rrset` also calls `emit_pre_sig` inside an `assert!`. So I did the same, here.
As both functions return a result, just forwarding an error (as improbable as it is) is an option.